hive/src/main/java/io/confluent/connect/storage/hive/HiveSchemaConverter.java

```diff
@@ -90,6 +98,29 @@ public static TypeInfo convertMap(Schema schema) {
   }
 
   public static TypeInfo convertPrimitive(Schema schema) {
-    return TYPE_TO_TYPEINFO.get(schema.type());
+    if (schema.name() == null) {
+      return TYPE_TO_TYPEINFO.get(schema.type());
+    }
+
+    //TODO: should take Hive version into consideration
+    switch (schema.name()) {
+      case Decimal.LOGICAL_NAME:
+        String scale = schema.parameters().get(Decimal.SCALE_FIELD);
+        String precision = schema.parameters().getOrDefault(CONNECT_AVRO_DECIMAL_PRECISION_PROP,
```

Comment:
In other config properties you'll see examples of getting the default from a config that has a default defined.

Reply:
A note on the default value of precision: `AvroData` uses 64 as default, but Hive allows at most 38, so I use 38 as default.